### PR TITLE
GNOME polishing from Q.A findings

### PR DESCRIPTION
**Pull request by worldofpeace**
###### Motivation for this change
Some stuff I thought to fixup when rereading https://github.com/NixOS/nixpkgs/issues/99648

Especially the thing about allowing DE modules to register a flatpak GUI that will be installed automatically.
Though, it's of question if there should be an option to disable that?

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/desktops/flatpak.nix [hunk 1]
```diff
@@ -15,6 +15,17 @@ in {
   options = {
     services.flatpak = {
       enable = mkEnableOption "flatpak";
+
+      guiPackages = mkOption {
+        type = types.listOf types.package;
```
Comment (jtojnar) [inline nixos/modules/services/desktops/flatpak.nix:20]: Perhaps this should be internal. We might want to drop it when we finish the Nix PackageKit support.

Comment (worldofpeace) [reply nixos/modules/services/desktops/flatpak.nix:20]: Sounds good to make it internal, but I'm not sure why we should drop it if there's Nix PackageKit support

**Comment by worldofpeace** [comment]
It seems https://github.com/NixOS/nixpkgs/pull/101516/commits/bcd186fb143c10e4516a1752530f4f60070c64bb did nothing... maybe we need an internal option for favorite apps

File: nixos/modules/installer/cd-dvd/installation-cd-graphical-gnome.nix [hunk 2]
```diff
@@ -9,7 +9,15 @@ with lib;
...
   isoImage.edition = "gnome";
 
-  services.xserver.desktopManager.gnome3.enable = true;
+  services.xserver.desktopManager.gnome3 = {
+    enable = true;
+
+    # Add firefox to shell favorites
+    extraGSettingsOverrides = ''
+      [ org.gnome.shell ]
+      favorite-apps=[ 'firefox.desktop' 'org.gnome.Geary.desktop', 'org.gnome.Music.desktop', 'org.gnome.Photos.desktop', 'org.gnome.Nautilus.desktop', 'org.gnome.Software.desktop' ]
```
Comment (jtojnar) [inline nixos/modules/installer/cd-dvd/installation-cd-graphical-gnome.nix:18]:
```suggestion
      [org.gnome.shell]
      favorite-apps=[ 'firefox.desktop'. 'org.gnome.Geary.desktop', 'org.gnome.Music.desktop', 'org.gnome.Photos.desktop', 'org.gnome.Nautilus.desktop', 'org.gnome.Software.desktop' ]
```

**Comment by worldofpeace** [comment]
I had to add an option for favorite-apps, but it's internal since it's only for the installation-cd, and it would be better suited to actually have a better dconf module.

**Comment by worldofpeace** [comment]
Also noticed that gnome-calendar should be in favorites as well https://github.com/NixOS/nixpkgs/pull/101516/commits/7df6af303e78986fa208c84f349afd928874fe62

**Comment by jtojnar** [comment]
It should not really matter that the option was there twice, I think the issue were the syntax error (missing comma).

**Comment by worldofpeace** [comment]
> It should not really matter that the option was there twice, I think the issue were the syntax error (missing comma).

I corrected it to no effect. Maybe I should check again, but I believe I tried building vms with clean states each time.

**Pull request merged**